pimd/pim_ifchannel.c

```diff
@@ -174,7 +174,10 @@ void pim_ifchannel_delete(struct pim_ifchannel *ch)
 	   ifchannel list is empty before deleting upstream_del
 	   ref count will take care of it.
 	*/
-	pim_upstream_del(pim_ifp->pim, ch->upstream, __PRETTY_FUNCTION__);
+	if (ch->upstream->ref_count > 0)
+		pim_upstream_del(pim_ifp->pim, ch->upstream,
+			__PRETTY_FUNCTION__);
+
```

Comment:
Can we put an else here with a zlog_err to describe what we are avoiding so that we can attempt to debug/fix this properly in the future